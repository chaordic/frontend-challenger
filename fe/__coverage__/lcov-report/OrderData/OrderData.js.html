<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for OrderData/OrderData.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">OrderData</a> OrderData.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>20/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { object, array } from 'prop-types';
&nbsp;
const OrderData = ({ payments, totals }) =&gt; (
  &lt;section&gt;
    &lt;h1&gt;Dados do pagamento&lt;/h1&gt;
    {totals !== undefined &amp;&amp;
      &lt;div&gt;
        {totals.subtotal !== undefined &amp;&amp;
          &lt;p&gt;Subtotal: {totals.subtotal}&lt;/p&gt;
        }
        {totals.freightCosts !== undefined &amp;&amp;
          &lt;p&gt;Frete: {totals.freightCosts}&lt;/p&gt;
        }
        {totals.discount !== undefined &amp;&amp;
          &lt;p&gt;Desconto: {totals.discount}&lt;/p&gt;
        }
        {totals.discount !== undefined &amp;&amp;
          &lt;p&gt;Valor total: {totals.discount}&lt;/p&gt;
        }
      &lt;/div&gt;
    }
    &lt;hr /&gt;
&nbsp;
    &lt;div&gt;
      &lt;h1&gt;MÃ©todo de pagamento&lt;/h1&gt;
      {payments !== undefined &amp;&amp; payments.map((payment, i) =&gt; (
        &lt;p key={i}&gt;
          {payment.brand !== undefined &amp;&amp;
            payment.brand
          }
&nbsp;
          {payment.number !== undefined &amp;&amp;
            payment.number
          }
&nbsp;
          {payment.expiresAt !== undefined &amp;&amp;
            ` Exp. ${payment.expiresAt} `
          }
&nbsp;
          {payment.installments !== undefined &amp;&amp;
            ` ${payment.installments}x de R$${payment.amount}`
          }
        &lt;/p&gt;
      ))}
    &lt;/div&gt;
&nbsp;
  &lt;/section&gt;
);
&nbsp;
OrderData.propTypes = {
  payments: array,
  totals: object,
};
&nbsp;
OrderData.defaultProps = {
  payments: [],
  totals: {},
};
&nbsp;
export default OrderData;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 11 2019 23:02:12 GMT+0000 (WET)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
