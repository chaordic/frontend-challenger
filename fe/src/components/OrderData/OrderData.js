import React from 'react';
import { object, array } from 'prop-types';

const OrderData = ({ payments, totals }) => (
  <section>
    <h1>Dados do pagamento</h1>
    {totals !== undefined &&
      <div>
        {totals.subtotal !== undefined &&
          <p>Subtotal: {totals.subtotal}</p>
        }
        {totals.freightCosts !== undefined &&
          <p>Frete: {totals.freightCosts}</p>
        }
        {totals.discount !== undefined &&
          <p>Desconto: {totals.discount}</p>
        }
        {totals.discount !== undefined &&
          <p>Valor total: {totals.discount}</p>
        }
      </div>
    }
    <hr />

    <div>
      <h1>MÃ©todo de pagamento</h1>
      {payments !== undefined && payments.map((payment, i) => (
        <p key={i}>
          {payment.brand !== undefined &&
            payment.brand
          }

          {payment.number !== undefined &&
            payment.number
          }

          {payment.expiresAt !== undefined &&
            ` Exp. ${payment.expiresAt} `
          }

          {payment.installments !== undefined &&
            ` ${payment.installments}x de R$${payment.amount}`
          }
        </p>
      ))}
    </div>

  </section>
);

OrderData.propTypes = {
  payments: array,
  totals: object,
};

OrderData.defaultProps = {
  payments: [],
  totals: {},
};

export default OrderData;
