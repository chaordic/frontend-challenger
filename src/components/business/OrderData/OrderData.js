import React from "react";
import PropTypes from "prop-types";
import { useSelector } from "react-redux";

import Value from "../../Value";
import withCard from "../../../hoc/withCard";
import dateFormat from "../../../lib/dateFormat";
import Flex from "../../Flex";

export const OrderData = ({ createdAt, pointOfSale }) => (
  <Flex>
    <Value label="Comprado em">
      {createdAt && dateFormat(createdAt, true)}
    </Value>
    <Value label="Ponto de Venda">{pointOfSale}</Value>
    <Value label="Modalidade de Entrega">
      F1 Envio pela loja, F2 Retira em Loja
      {/* TODO:Rever dados */}
    </Value>
  </Flex>
);

OrderData.propTypes = {
  createdAt: PropTypes.string,
  pointOfSale: PropTypes.string
};

export default props => {
  const { data, ...status } = useSelector(({ delivery }) => delivery);
  return withCard(OrderData)({
    title: "Dados do pedido",
    ...props,
    ...data,
    ...status
  });
};
