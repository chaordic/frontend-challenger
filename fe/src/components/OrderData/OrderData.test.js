import React from 'react';
import renderer from 'react-test-renderer';
import OrderData from './OrderData';

const props = {
  payments: [{
    amount: 'R$500,00',
    brand: 'VISA',
    expiresAt: '02/23',
    installments: '1',
    number: '**** **** **** 3455',
    type: 'CREDIT_CARD',
  }],
  totals: {
    isLoading: false,
    subtotal: 'R$500,00',
    freightCosts: 'R$20,00',
    discount: 'R$10,00',
    total: 'R$510,00',
  },
};

describe('(Component) OrderData', () => {
  it('should render an OrderData with result', () => {
    const tree = renderer.create(<OrderData {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
