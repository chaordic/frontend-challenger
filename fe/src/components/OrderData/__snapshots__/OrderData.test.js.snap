// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) OrderData should render an OrderData with result 1`] = `
<section>
  <h1>
    Dados do pagamento
  </h1>
  <div>
    <p>
      Subtotal: 
      R$500,00
    </p>
    <p>
      Frete: 
      R$20,00
    </p>
    <p>
      Desconto: 
      R$10,00
    </p>
    <p>
      Valor total: 
      R$10,00
    </p>
  </div>
  <hr />
  <div>
    <h1>
      MÃ©todo de pagamento
    </h1>
    <p>
      VISA
      **** **** **** 3455
       Exp. 02/23 
       1x de R$R$500,00
    </p>
  </div>
</section>
`;
